joss/paper.md

```diff
@@ -0,0 +1,118 @@
+are defined using a left-handed coordinate system of physical coordinates. In
+words from the neuroimaging domain, the coordinate system of transforms is
+_RAS+_ (or positive directions point to the Righthand for the first axis,
+Anterior for the second, and Superior for the third axis). The internal
+representation of transform coordinates is the most relevant design decision,
+and implies that a conversion of coordinate system is necessary to correctly
+interpret transforms generated by other software. When a transform that is
+defined in another coordinate system is loaded, it is automatically converted
+into _RAS+_ space.
+
+`nitransforms` was developed using a test-driven development paradigm, with the
+battery of tests being written prior to the software implementations. Two
+categories of tests were used: unit tests and cross-tool comparison tests. Unit
+tests evaluate the formal correctness of the implementation, while cross-tool
+comparison tests assess the correct implementation of third-party software.
+The testing suite is incorporated into a continuous integration framework, which
+assesses the continuity of the implementation along the development life and
+ensures that code changes and additions do not break existing functionalities.
+
+# Spatial transforms
+
+Let $\vec{x}$ represent the coordinates of a point in the reference coordinate
+system $R$, and $\vec{x}'$ its projection on to another coordinate system $M$:
+
+$T\colon R \subset \mathbb{R}^n \to M \subset \mathbb{R}^n$
+
+$\vec{x} \mapsto \vec{x}' = f(\vec{x}).$
+
+In an image registration problem, $M$ is a moving image from which we want to
+sample data in order to bring the image into spatial alignment with the
+reference image $R$. Hence, $f$ here is the spatial transformation function
+that maps from coordinates in $R$ to coordinates in $M$. There are a
+multiplicity of image registration algorithms and corresponding image
+transformation models to estimate linear and nonlinear transforms.
+
+The problem has been traditionally confused by the need of _transforming_ or
+mapping one image (generally referred to as _moving_) into another that serves
+as reference, with the goal of _fusing_ the information from both. An example of
+image fusion application would be the alignment of functional data from one
+individual's brain to the same individual's corresponding anatomical MRI scan
+for visualization. Therefore, "applying a transform" entails two operations:
+first, transforming the coordinates of the samples in the reference image $R$ to
+find their mapping $\vec{x}'$ on $M$ via $T\{\cdot\}$, and second, an
+interpolation step as $\vec{x}'$ will likely fall off-the-grid of the moving
+image $M$. These two operations are confusing because, while the spatial
+transformation projects from $R$ to $M$, the data flows in reversed way after the
+interpolation of the values of $M$ at the mapped coordinates $\vec{x}'$.
+
```

Comment:
I'd add a section of examples, copying from the preprint.

Reply:
their docs explicitly state
> software documentation such as API (Application Programming Interface) functionality should not be in the paper and instead should be outlined in the software documentation.

Reply:
I think an example or two cannot be considered software documentation, and they will very quickly convey what the tool does (ideally, I'm imagining two images on the left with some object signifying an input transform, they feed into the snippet and the output shows the moving image deformed via T). I can take a stab on this next week, don't worry about this.